### DISCUSSION TITLE
Concurrent requests with one resty client?

### DISCUSSION THREAD

**Discussion by jo-me** [go-resty/resty#556] [Q&A] [answered]
Can multiple go routines that make http requests (potentially concurrently) share the same client or do they need to create a new client for every request?

They share the same underlying transport / authentication  settings. They just call different endpoints of the same REST API.

**Comment by nicoxiang** [comment, marked as answer]
I asked a similar question before, see #542 , and i got the answer

> You only need one instance of the client which be used to create any number of requests

And as described in the README Resty design section

> goroutine concurrent safe

In the resty source code,  `resty.New()` function will create golang standard [`http.Client`](https://pkg.go.dev/net/http#Client)  , and the docs also said:

> The Client's Transport typically has internal state (cached TCP connections), so Clients should be reused instead of created as needed. Clients are safe for concurrent use by multiple goroutines.

Related issue: #328
